### PR TITLE
[Beta] Updated the documentation for useCallback

### PR DESCRIPTION
**Pull request by arati-1**
Created a draft for useCallback and updated the `sideBarReference.json`.

## [Preview](https://beta-reactjs-org-git-fork-arati-1-api-usecallback-fbopensource.vercel.app/apis/react/useCallback)

### CODE REVIEW

File: beta/src/pages/apis/react/useCallback.md [hunk 1]
```diff
@@ -2,17 +2,278 @@
...
+
+```js
+const memoizedCallback = useCallback(callback, [...dependencies])
+```
 
-This section is incomplete, please see the old docs for [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback).
+</Intro>
 
-</Wip>
+<Note>
```
Comment (rickhanlonii) [inline beta/src/pages/apis/react/useCallback.md:15]: Is this worth adding to the `useMemo` docs too?

File: beta/src/pages/apis/react/useCallback.md [hunk 2]
```diff
@@ -2,17 +2,278 @@
...
+
+## Usage {/*usage*/}
+
+### Skip re-rendering of the components {/*skip-re-rendering-of-the-components*/}
+
+Often you come across scenarios where certain components render needlessly every time a button is clicked or an input changes. This unnecessary re-render on every change can be an impediment to an optimized application. `useCallback` hook provided by React can help in optimization and skipping unnecessary re-renders.
+
+Consider the following example:
+
+Call `useCallback` at the top level of your component to use it.
```
Comment (rickhanlonii) [inline beta/src/pages/apis/react/useCallback.md:38]: Should we also add CodeSteps here (similar feedback to useMemo)?

File: beta/src/pages/apis/react/useCallback.md [hunk 3]
```diff
@@ -2,17 +2,278 @@
...
 
 ```js
-const memoizedCallback = useCallback(callback, [...dependencies])
+import { useCallback } from 'react';
+  // ...
 ```
+In this example, the `Next` button renders every time the `Show details` button is clicked, even though the conditions for it are unchanged. The event handling function `handleNext`  is created every time the components render.
 
-</Intro>
+``` js [[6-7],20]
```
Comment (rickhanlonii) [inline beta/src/pages/apis/react/useCallback.md:46]: This are pretty big snippets, should they be Sandpacks?

File: beta/src/pages/apis/react/useCallback.md (hunk 1 above)
Comment (arati-1) [reply beta/src/pages/apis/react/useCallback.md:15]: Will add.

**Pull request closed without merging**